### PR TITLE
MINOR: Propagate LogContext to channel builders and SASL authenticator

### PR DESCRIPTION
**Pull request by hachikuji**
The log context is useful when debugging applications which have multiple clients. This patch propagates the context to the channel builders and the SASL authenticator.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: clients/src/main/java/org/apache/kafka/common/network/ChannelBuilders.java [hunk 1]
```diff
@@ -45,18 +46,21 @@ private ChannelBuilders() { }
      * @param config client config
      * @param listenerName the listenerName if contextType is SERVER or null otherwise
      * @param clientSaslMechanism SASL mechanism if mode is CLIENT, ignored otherwise
+     * @param time the time instance
      * @param saslHandshakeRequestEnable flag to enable Sasl handshake requests; disabled only for SASL
      *             inter-broker connections with inter-broker protocol version < 0.10
      * @return the configured `ChannelBuilder`
```
Comment (rondagostino) [inline clients/src/main/java/org/apache/kafka/common/network/ChannelBuilders.java:54]: needs `@param logContext`

File: clients/src/main/java/org/apache/kafka/clients/ClientUtils.java [hunk 2]
```diff
@@ -93,11 +94,11 @@ private ClientUtils() {
      * @param config client configs
      * @return configured ChannelBuilder based on the configs.
```
Comment (rondagostino) [inline clients/src/main/java/org/apache/kafka/clients/ClientUtils.java:100]: Needs `@param logContext` (and also `@param time the time instance`)

File: clients/src/main/java/org/apache/kafka/common/network/ChannelBuilders.java [hunk 3]
```diff
@@ -81,9 +85,11 @@ public static ChannelBuilder serverChannelBuilder(ListenerName listenerName,
                                                       AbstractConfig config,
                                                       CredentialCache credentialCache,
                                                       DelegationTokenCache tokenCache,
-                                                      Time time) {
+                                                      Time time,
+                                                      LogContext logContext) {
```
Comment (rondagostino) [inline clients/src/main/java/org/apache/kafka/common/network/ChannelBuilders.java:96]: Needs `@param logContext` in Javadoc (and also `@param isInterBrokerListener`, `@param isInterBrokerListener `, and `@param time`)

File: clients/src/main/java/org/apache/kafka/common/network/SaslChannelBuilder.java [hunk 4]
```diff
@@ -56,8 +57,9 @@
 import org.ietf.jgss.GSSName;
 import org.ietf.jgss.Oid;
 import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 
+import javax.security.auth.Subject;
```
Comment (rondagostino) [inline clients/src/main/java/org/apache/kafka/common/network/SaslChannelBuilder.java:61]: These `javax...` imports generally appear in a separate section afterwards; any reason to move them up here?

**Review by rondagostino — APPROVED** [review]
LGTM, just some minor Javadoc comments

**Comment by rondagostino** [comment]
Actually, the compile failed; it looks like there is an additional change needed:

```
01:39:50 > Task :tools:compileJava FAILED
01:39:50 /home/jenkins/jenkins-slave/workspace/kafka-pr-jdk11-scala2.12/tools/src/main/java/org/apache/kafka/trogdor/workload/ConnectionStressWorker.java:156: error: method createChannelBuilder in class ClientUtils cannot be applied to given types;
01:39:50             this.channelBuilder = ClientUtils.createChannelBuilder(conf, TIME);
01:39:50                                              ^
01:39:50   required: AbstractConfig,Time,LogContext
01:39:50   found: AdminClientConfig,Time
01:39:50   reason: actual and formal argument lists differ in length
01:39:50 1 error
```

**Comment by hachikuji** [comment]
Thanks @rondagostino. Addressed comments.

**Comment by hachikuji** [comment]
retest this please

**Comment by rondagostino** [comment]
Looks like there is another place to adjust. I did a quick search and this should be the last one.

```
20:50:01 > Task :connect:runtime:compileJava FAILED
20:50:01 /home/jenkins/jenkins-slave/workspace/kafka-pr-jdk8-scala2.12/connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/WorkerGroupMember.java:101: error: method createChannelBuilder in class ClientUtils cannot be applied to given types;
20:50:01             ChannelBuilder channelBuilder = ClientUtils.createChannelBuilder(config, time);
20:50:01                                                        ^
20:50:01   required: AbstractConfig,Time,LogContext
20:50:01   found: DistributedConfig,Time
20:50:01   reason: actual and formal argument lists differ in length
20:50:01 1 error
```

**Review by omkreddy — APPROVED** [review]
@hachikuji Thanks for the PR. LGTM after addressing @rondagostino's comments.

**Pull request merged**